rswag-specs/lib/rswag/specs/openapi_formatter.rb

```diff
@@ -0,0 +1,259 @@
+
+        verb = metadata[:operation][:verb]
+        operation = metadata[:operation]
+          .reject { |k, _v| k == :verb }
+          .merge(responses: { response_code => response })
+
+        path_template = metadata[:path_item][:template]
+        path_item = metadata[:path_item]
+          .reject { |k, _v| k == :template }
+          .merge(verb => operation)
+
+        { paths: { path_template => path_item } }
+      end
+
+      def doc_version(doc)
+        doc[:openapi]
+      end
+
+      def upgrade_response_produces!(openapi_spec, metadata)
+        # Accept header
+        mime_list = Array(metadata[:operation][:produces] || openapi_spec[:produces])
+        target_node = metadata[:response]
+        upgrade_content!(mime_list, target_node)
+        metadata[:response].delete(:schema)
+      end
+
+      def upgrade_content!(mime_list, target_node)
+        schema = target_node[:schema]
+        return if mime_list.empty? || schema.nil?
+
+        target_node[:content] ||= {}
+        mime_list.each do |mime_type|
+          # TODO: upgrade to have content-type specific schema
+          (target_node[:content][mime_type] ||= {}).merge!(schema: schema)
+        end
+      end
+
+      def upgrade_request_type!(metadata)
+        # No deprecation here as it seems valid to allow type as a shorthand
+        operation_nodes = metadata[:operation][:parameters] || []
+        path_nodes = metadata[:path_item][:parameters] || []
+        header_node = metadata[:response][:headers] || {}
+
+        (operation_nodes + path_nodes + [header_node]).each do |node|
+          if node && node[:type] && node[:schema].nil?
+            node[:schema] = { type: node[:type] }
+            node.delete(:type)
+          end
+        end
+      end
+
+      def remove_invalid_operation_keys!(value)
+        return unless value.is_a?(Hash)
+
+        value.delete(:consumes) if value[:consumes]
+        value.delete(:produces) if value[:produces]
+        value.delete(:request_examples) if value[:request_examples]
+        value[:parameters].each { |p| p.delete(:getter) } if value[:parameters]
+      end
+
+      def parse_parameters_to_oas3_syntax(doc)
+        doc[:paths]&.each_pair do |_k, path|
+          path.each_pair do |_verb, endpoint|
+            is_hash = endpoint.is_a?(Hash)
+            if is_hash && endpoint[:parameters]
+              mime_list = endpoint[:consumes] || doc[:consumes]
+              parse_parameters(endpoint, mime_list) if mime_list
+            end
+            remove_invalid_operation_keys!(endpoint)
+          end
+        end
+      end
+
+      def parse_parameters(endpoint, mime_list)
+        parameters = endpoint[:parameters]
+        # There can only be 1 body parameter in Swagger 2.0, so while in OAS3 we interpret
+        # body parameters as formData, only consider the first body we encounter.
+        schema_param = parameters.find { |p| (p[:in] == :body) && p[:schema] }
```

Comment:
Thinking of the readability here:

```suggestion
        schema_param = parameters.find { |p| p[:schema] if p[:in] == :body }
```